server/infrastructure/auth_repository.py

```
@@ -0,0 +1,132 @@
+    async def verify_custom_token(self, token: str) -> Dict[str, str]:
+        """Verify custom token"""
+        try:
+            # Decode without verification first to get the uid
+            decoded = jwt.decode(token, options={"verify_signature": False})
+            uid = decoded.get('uid')
+            
+            if not uid:
+                raise HTTPException(
+                    status_code=status.HTTP_401_UNAUTHORIZED,
+                    detail="Invalid token format"
+                )
```

Ruff (0.14.1)

Abstract `raise` to an inner function

(TRY301)